fixtures/shared/src/mock-console.ts

```diff
@@ -0,0 +1,60 @@
+import * as util from "node:util";
+import { afterEach, beforeEach, vi } from "vitest";
+import type { MockInstance } from "vitest";
+
+/**
+ * We use this module to mock console methods, and optionally
+ * assert on the values they're called with in our tests.
+ */
+
+let debugSpy: MockInstance,
+	logSpy: MockInstance,
+	infoSpy: MockInstance,
+	errorSpy: MockInstance,
+	warnSpy: MockInstance;
+
+const std = {
+	get debug() {
+		return normalizeOutput(debugSpy);
+	},
+	get out() {
+		return normalizeOutput(logSpy);
+	},
+	get info() {
+		return normalizeOutput(infoSpy);
+	},
+	get err() {
+		return normalizeOutput(errorSpy);
+	},
+	get warn() {
+		return normalizeOutput(warnSpy);
+	},
+};
+
+function normalizeOutput(spy: MockInstance, join = "\n"): string {
+	return captureCalls(spy, join);
+}
+
+function captureCalls(spy: MockInstance, join = "\n"): string {
+	return spy.mock.calls
+		.map((args: unknown[]) => util.format("%s", ...args))
+		.join(join);
+}
```

Comment:
Nit: What do we need an extra `captureCalls` helper here? I don't see it being used outside of `normalizeOutput`.

Reply:
Are you suggesting it should be inline in `normalizeOutput()`?

Reply:
Yes. But I guess you are just following the patterns. nevermind!